src/AspNet.Security.OAuth.Apple/AppleAuthenticationExtensions.cs

```diff
@@ -76,15 +74,14 @@ public static AuthenticationBuilder AddApple([NotNull] this AuthenticationBuilde
             [CanBeNull] string caption,
             [NotNull] Action<AppleAuthenticationOptions> configuration)
         {
+            builder.Services.AddMemoryCache();
             builder.Services.TryAddSingleton<AppleClientSecretGenerator, DefaultAppleClientSecretGenerator>();
```

Comment:
Should we make all these services options while we're at it? (to allow folks have different implementations per instance)

Reply:
Implementers can already do that by overriding the DI registrations before calling `AddApple()` because the `TryAdd...()` methods are used.

Making them options would make it more explicit to override what is in DI, but we'd still need to register them with DI to create the defaults (because they need the clock, a logger etc.).

Given no-one has asked for this functionality in the last year, I don't think we need to further bloat the options with another way of doing something that's already possible.

Reply:
> Implementers can already do that by overriding the DI registrations before calling AddApple() because the TryAdd...() methods are used.

Not really. Services are shared between all schemes/instances, so you can't have one `AppleClientSecretGenerator` per Apple authentication handler instance (e.g `scheme1` and `scheme2`).

> Making them options would make it more explicit to override what is in DI, but we'd still need to register them with DI to create the defaults (because they need the clock, a logger etc.).

No, you can instantiate the "services-as-options" in an `IPostConfigureOptions` class, which is a very classical pattern in ASP.NET Core (e.g https://github.com/dotnet/aspnetcore/blob/master/src/Security/Authentication/OpenIdConnect/src/OpenIdConnectPostConfigureOptions.cs#L35).

That said, if you think it's not worth it, that's fine.

Reply:
Yeah I was referencing [the OAuth version](https://github.com/dotnet/aspnetcore/blob/ab973d23fb5fc3b00d37ba63182b5bb14dc02803/src/Security/Authentication/OAuth/src/OAuthPostConfigureOptions.cs#L33) where you can provide a data protection provider, but it falls back to the one in DI (not doing a `new ...()` where you need to get hold of all of _those_ things' dependencies too).

Reply:
This is now all handled in `ApplePostConfigureOptions`.